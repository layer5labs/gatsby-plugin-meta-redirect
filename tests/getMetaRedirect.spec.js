const getMetaRedirect = require('../getMetaRedirect');

describe('getMetaRedirect', () => {
  it('wraps path in forward slashes', () => {
    expect(getMetaRedirect('toPath')).toMatchSnapshot();
  });

  it('allows existing leading and trailing forward slashes', () => {
    expect(getMetaRedirect('/toPath/')).toMatchSnapshot();
  });

  it('trims leading and trailing whitespace', () => {
    expect(getMetaRedirect(' toPath ')).toMatchSnapshot();
  });

  it('handles deep paths', () => {
    expect(getMetaRedirect('a/b/c/d')).toMatchSnapshot();
  });

  it('handles offset wrapping forward slashes', () => {
    expect(getMetaRedirect('a/b/c/')).toMatchSnapshot();
  });

  it('replaces duplicate slashes with single slash', () => {
    expect(getMetaRedirect('topath//a')).toMatchSnapshot();
  });

  it('leaves full urls untouched', () => {
    expect(getMetaRedirect('http://example.com')).toMatchSnapshot();
    expect(getMetaRedirect('http://example.com/')).toMatchSnapshot();
    expect(getMetaRedirect('http://example.com/a/b/c')).toMatchSnapshot();
  });

  it('handles redirecting to root', () => {
    expect(getMetaRedirect('/')).toMatchSnapshot();
  });

  it('handles redirecting to a file', () => {
    expect(getMetaRedirect('/test.txt')).toMatchSnapshot();
  });

  it('handles redirecting to a file in a folder', () => {
    expect(getMetaRedirect('a/b/test.txt')).toMatchSnapshot();
  });
});
