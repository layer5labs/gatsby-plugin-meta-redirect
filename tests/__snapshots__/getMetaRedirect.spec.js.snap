// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMetaRedirect allows existing leading and trailing forward slashes 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/toPath/'\\" />"`;

exports[`getMetaRedirect handles deep paths 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/a/b/c/d/'\\" />"`;

exports[`getMetaRedirect handles offset wrapping forward slashes 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/a/b/c/'\\" />"`;

exports[`getMetaRedirect handles redirecting to a file 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/test.txt'\\" />"`;

exports[`getMetaRedirect handles redirecting to a file in a folder 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/a/b/test.txt'\\" />"`;

exports[`getMetaRedirect handles redirecting to root 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/'\\" />"`;

exports[`getMetaRedirect leaves full urls untouched 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='http://example.com'\\" />"`;

exports[`getMetaRedirect leaves full urls untouched 2`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='http://example.com/'\\" />"`;

exports[`getMetaRedirect leaves full urls untouched 3`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='http://example.com/a/b/c'\\" />"`;

exports[`getMetaRedirect replaces duplicate slashes with single slash 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/topath/a/'\\" />"`;

exports[`getMetaRedirect trims leading and trailing whitespace 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/toPath/'\\" />"`;

exports[`getMetaRedirect wraps path in forward slashes 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/toPath/'\\" />"`;
